### PR TITLE
Parse microseconds in string input to `unix!`

### LINKED ISSUE #29: Microsecond precision is too high on Calendar.DateTime.Parse.unix!
I found that `Calendar.DateTime.Parse.unix!/1` always sets precision to 6 when given a float.

```
dt = Calendar.DateTime.Parse.unix! 1_000_000_000.123
dt.microsecond #=> {123000, 6}
```

I would have expected precision to either match the float I'm supplying (i.e. `1.1000` has precision 4) or the number of significant digits in the decimal part (i.e. `1.1000` has precision 1). Possibly default to significant digits (seems safer not to assume precision), and accept an option like `precise: true` to keep the full precision (after rounding to nearest millionth).

I can help out with the implementation if you're swamped, but want the change.

PS. Thanks for the great work.

### PR DESCRIPTION
**Pull request by martinsvalin**
Vacation time got in the way, so I took a good while to get back to the microseconds parsing.
This closes #29 as far as I'm concerned.

There's a doctest showing that `unix!` can parse microsecond fractions in string input. I put an edge case for capping the precision in the test file because I didn't want to clutter the docs with low-value examples.

I found that `parse_fraction` would truncate rather than round input with a long fraction. This looked like a bug, so I fixed it. This private function seemed to cover what `parse_unix_fraction` did. Since floats are turned into a 6-digit string, precision will be 6, but that's a subtle effect of the implementation, so I added a test for this as well.

### CODE REVIEW

File: test/date_time_parse_test.exs [hunk 1]
```diff
@@ -3,5 +3,14 @@ defmodule DateTimeParseTest do
   alias Calendar.DateTime.Parse
   import Parse
   doctest Parse
-end
 
+  test "microsecond precision is capped at 6" do
+    assert(unix!("1000000000.123456789").microsecond == {123457, 6})
```
Comment (lau) [inline test/date_time_parse_test.exs]: Thanks, in general it looks good. For consistency we should not round up, could you do it so that this will return {123456, 6} instead? An issue with rounding is that e.g. 1000000000.99999999 becomes 1000000001.0. Rounding could change the year in some cases.

Comment (martinsvalin) [inline test/date_time_parse_test.exs]: Good catch. It's even worse, since the PR as it stands now would round that to `1000000`, and set `microsecond: {1000000, 6}`. That's not a valid microsecond part. I'll revert back to the truncating behaviour.

**Comment by lau** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
